usr/src/cmd/format/io.c

```diff
@@ -691,7 +692,7 @@ input(int type, char *promptstr, int delim, u_ioparam_t *param, int *deflt,
 				/*
 				 * Duplicate and return the default string
 				 */
-				return ((int)alloc_string((char *)deflt));
+				return ((uint64_t)alloc_string((char *)deflt));
```

Comment:
If we're going to do this -- and looking at the code, we have to -- this cast needs to be to `intptr_t` and `input()` needs to return it.

(and if they're going to use deflt like a `void *` they really should declare it that way, but that's probably a mess to fix now.

Reply:
I tried addressing both see if you can live with the result.